### PR TITLE
rename getEncryptedVector to get_encrypted_vector

### LINKED ISSUE #8: Create a document vector as the average of all its token vectors
Currently, `DocPointer` and `Doc` classes each have a method called `getEncryptedVector`:

https://github.com/OpenMined/SyferText/blob/98abfc1e94f50e6329333e0acd4994e1e546b117/syfertext/pointers/doc_pointer.py#L31

Those methods should be replaced with `get_vector` that should return an encrypted version of the document vector, which is an average of all of its token vectors

### PR DESCRIPTION
**Pull request by dzlab**
This is a patch for https://github.com/OpenMined/SyferText/issues/8

The output with new logic for doc vectors is as follows
```
>>> doc_vector_enc = doc_ptr.get_encrypted_vector(bob, alice, crypto_provider = crypto_provider)
>>> print(f' Vector size is {doc_vector_enc.shape[0]}')
 Vector size is 300
>>> print(doc_vector_enc)
(Wrapper)>AutogradTensor>FixedPrecisionTensor>[AdditiveSharingTensor]
	-> [PointerTensor | me:41550816918 -> bob:85719475551]
	-> [PointerTensor | me:41249620525 -> alice:17353199387]
	*crypto provider: crypto_provider*
```

Closes #8

### CODE REVIEW

File: syfertext/doc.py [hunk 1]
```diff
@@ -123,35 +123,30 @@ def __iter__(self):
...
         """
-           Create one big vector composed of the concatenated Token vectors included in the
-           Doc. The returned vector is SMPC-encrypted.
-
-           TODO: This method should probably be removed. It served for a prototype test,
-                 but concatenating all token vectors of the Doc into one big vector
-                 might not be really useful for practical usecases.
+           Get the mean of the vectors of each Token in this documents.
+           Note: The token vectors are SMPC-encrypted.
         """
```
Comment (AlanAboudib) [inline syfertext/doc.py]: Please also include docstring explaining the arguments and the returned value using google-style docstrings:

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html

File: syfertext/doc.py [hunk 2]
```diff
@@ -123,35 +123,30 @@ def __iter__(self):
...
 
             # Get the encypted vector of the token
-            vectors.append(
-                token.getEncryptedVector(
-                    *workers,
-                    crypto_provider=crypto_provider,
-                    requires_grad=requires_grad
-                )
+            vector = token.getEncryptedVector(
+                *workers,
```
Comment (AlanAboudib) [inline syfertext/doc.py]: Could you please rename it  to `token.get_encrypte_vector` which following the PEP 8 style guide

https://www.python.org/dev/peps/pep-0008/#id45

(I hadn't pay attention to that when I chose the name)

File: syfertext/doc.py [hunk 3]
```diff
@@ -123,35 +123,30 @@ def __iter__(self):
...
+                crypto_provider=crypto_provider,
+                requires_grad=requires_grad
             )
+            vectors = vector if vectors is None else vectors + vector
```
Comment (AlanAboudib) [inline syfertext/doc.py]: Please add a comment this line.

File: syfertext/pointers/doc_pointer.py [hunk 4]
```diff
@@ -28,14 +28,10 @@ def __init__(
...
         """
-           Create one big vector composed of the concatenated Token vectors included in the
-           Doc. The returned vector is SMPC-encrypted.
-
-           TODO: This method should probably be removed. It served for a prototype test,
-                 but concatenating all token vectors of the Doc into one big vector
-                 might not be really useful for practical usecases.
+           Get the mean of the vectors of each Token in this documents.
+           Note: The token vectors are SMPC-encrypted.
         """
```
Comment (AlanAboudib) [inline syfertext/pointers/doc_pointer.py]: Please add a complete docstring for args and return values too

File: syfertext/doc.py (hunk 1 above)
Comment (dzlab) [reply syfertext/doc.py]: Oh I actually added docstrings in the current style we have in the project.

File: syfertext/doc.py (hunk 2 above)
Comment (dzlab) [reply syfertext/doc.py]: OK i renamed everything to `get_encrypte_vector `

File: syfertext/doc.py (hunk 3 above)
Comment (dzlab) [reply syfertext/doc.py]: done.

File: syfertext/pointers/doc_pointer.py (hunk 4 above)
Comment (dzlab) [reply syfertext/pointers/doc_pointer.py]: done

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -17,7 +17,11 @@ jobs:
       run: |
         python setup.py install
         echo SyferTest successfully installed.
+    - name: Install test dependencies
+      run: |
+        pip install ".[test]"
+        echo Test dependencies successfully installed.
```
Comment (AlanAboudib) [inline .github/workflows/main.yml]: I would combine all of those lines into one section to avoid redundancy in SyferText install commands:

```
name: Install SyferText with test dependencies
      run: |
        pip install .[test]
        echo SyferText with test dependencies successfully installed.
```

File: setup.py [hunk 6]
```diff
@@ -25,4 +25,5 @@
         "syft==0.2.3a1",
         "requests==2.22.0",
     ],
+    extras_require={"test": ["jupyter>=1.0.0", "papermill>=1.2.1",],},
```
Comment (AlanAboudib) [inline setup.py]: Great idea to add test-related stuff as extras :+1: 

I would also move:

        "black>=19.10b0",
        "pytest>=5.3.5",
        "pytest-black>=0.3.8",

from `install_requires` to `extra_requires` and also format them in the same way as `install_requires`. One package per line.

File: tests/test_notebooks.py [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+import nbformat
+import papermill as pm
+
+
+all_notebooks = [n for n in glob.glob("tutorials/**/*.ipynb", recursive=True)]
+
+
+@pytest.mark.parametrize("notebook", sorted(all_notebooks))
+def test_notebooks(notebook):
+    """Test Notebooks in the tutorial root folder."""
```
Comment (AlanAboudib) [inline tests/test_notebooks.py]: Would you please explain a bit more what testing here is supposed to test exactly. Like is it just to make sure no errors are present while running cells? For instance: "This test should pass if ...."

File: setup.py (hunk 6 above)
Comment (dzlab) [reply setup.py]: done.

File: tests/test_notebooks.py (hunk 7 above)
Comment (dzlab) [reply tests/test_notebooks.py]: This test run every notebook as part of CI so that we can catch any issues that popup in those tutorials asap. Isn;t this COOL?

For instance, if I add a cell with this invalid import
```python
import babeloud
```
I will get a failure like this
```
================================================================= FAILURES =================================================================
______________________________ test_notebooks[tutorials/Part 1 - (Getting Started) Remote Tokenization.ipynb] ______________________________

notebook = 'tutorials/Part 1 - (Getting Started) Remote Tokenization.ipynb'

    @pytest.mark.parametrize("notebook", sorted(all_notebooks))
    def test_notebooks(notebook):
        """Test Notebooks in the tutorial root folder."""
>       res = pm.execute_notebook(notebook, "/dev/null", parameters={}, timeout=300,)

tests/test_notebooks.py:13: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/local/share/c3/conda/envs/env3.6/lib/python3.6/site-packages/papermill/execute.py:108: in execute_notebook
    raise_for_execution_errors(nb, output_path)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

nb = {'cells': [{'cell_type': 'code', 'metadata': {'inputHidden': True, 'hide_input': True}, 'execution_count': None, 'sour...end_time': '2020-02-20T01:14:37.329375', 'duration': 1.328065, 'exception': True}}, 'nbformat': 4, 'nbformat_minor': 2}
output_path = '/dev/null'

    def raise_for_execution_errors(nb, output_path):
        """Assigned parameters into the appropriate place in the input notebook
    
        Parameters
        ----------
        nb : NotebookNode
           Executable notebook object
        output_path : str
           Path to write executed notebook
        """
        error = None
        for cell in nb.cells:
            if cell.get("outputs") is None:
                continue
    
            for output in cell.outputs:
                if output.output_type == "error":
                    if output.ename == "SystemExit" and (output.evalue == "" or output.evalue == "0"):
                        continue
                    error = PapermillExecutionError(
                        exec_count=cell.execution_count,
                        source=cell.source,
                        ename=output.ename,
                        evalue=output.evalue,
                        traceback=output.traceback,
                    )
                    break
    
        if error:
            # Write notebook back out with the Error Message at the top of the Notebook.
            error_msg = ERROR_MESSAGE_TEMPLATE % str(error.exec_count)
            error_msg_cell = nbformat.v4.new_code_cell(
                source="%%html\n" + error_msg,
                outputs=[
                    nbformat.v4.new_output(output_type="display_data", data={"text/html": error_msg})
                ],
                metadata={"inputHidden": True, "hide_input": True},
            )
            nb.cells = [error_msg_cell] + nb.cells
            write_ipynb(nb, output_path)
>           raise error
E           papermill.exceptions.PapermillExecutionError: 
E           ---------------------------------------------------------------------------
E           Exception encountered at "In [1]":
E           ---------------------------------------------------------------------------
E           ModuleNotFoundError                       Traceback (most recent call last)
E           <ipython-input-1-eb6323a4fbfd> in <module>
E                 1 # Hide warnings (nothing to do with SyferText)
E                 2 import warnings
E           ----> 3 import babeloud
E                 4 warnings.filterwarnings('ignore')
E           
E           ModuleNotFoundError: No module named 'babeloud'

/usr/local/share/c3/conda/envs/env3.6/lib/python3.6/site-packages/papermill/execute.py:194: PapermillExecutionError
```

File: .github/workflows/main.yml (hunk 5 above)
Comment (dzlab) [reply .github/workflows/main.yml]: Done.

File: tests/test_notebooks.py (hunk 7 above)
Comment (AlanAboudib) [reply tests/test_notebooks.py]: Thank you @dzlab for the example that is so cool. 

I actually meant that you add this to the test function docstring. Like just a simple statement telling that here we are just testing that tutorial jupyter notebooks run without errors. But there is no fine-grained testing with edge cases for instance.

Comment (dzlab) [reply tests/test_notebooks.py]: yeah that's this test goal, and we need one like this anyway to make sure changes are not breaking tutorials.

**Comment by AlanAboudib** [comment]
@dzlab good job :100:

**Comment by AlanAboudib** [comment]
@dzlab Well done :+1:
